bin/yaml_to_json.py

```diff
@@ -0,0 +1,275 @@
+                            bas_det[detection] = [baseline]
+                        else:
+                            bas_det[detection].append(baseline)
+        return bas_det
+
+
+    def map_detection_to_story(self, detections):
+        det_sto = dict()
+        for detection in detections:
+            if 'tags' in detection:
+                if 'analytic_story' in detection['tags']:
+                    for story in detection['tags']['analytic_story']:
+                        if not (story in det_sto):
+                            det_sto[story] = [detection]
+                        else:
+                            det_sto[story].append(detection)
+        return det_sto
+
+
+    def generate_macro_dict(self, macros):
+        macro_dict = {}
+        for macro in macros:
+            macro_dict[macro['name']] = macro
+
+        return macro_dict
+
+    def generate_lookup_dict(self, lookups):
+        lookup_dict = {}
+        for lookup in lookups:
+            lookup_dict[lookup['name']] = lookup
+
+        return lookup_dict
+
+
+    def parse_and_add_macros(self, object):
+        macros_found = re.findall('\`([^\s]+)`', object['search'])
+        macros_filtered = set()
+        for macro in macros_found:
+            if not 'cim_' in macro and not 'get_' in macro and not '_filter' in macro and not 'drop_dm_object_name' in macro:
+                start = macro.find('(')
+                if start != -1:
+                    macros_filtered.add(macro[:start])
+                else:
+                    macros_filtered.add(macro)
+
+        macro_objects = []
+        for macro in list(macros_filtered):
+            lookups = self.parse_and_add_lookups(self.macros[macro]['definition'])
+            if len(lookups) > 0:
+                self.macros[macro]['lookups'] = lookups
+            macro_objects.append(self.macros[macro])
+
+        new_dict = {}
+        new_dict['definition'] = 'search *'
+        new_dict['description'] = 'Update this macro to limit the output results to filter out false positives. '
+        new_dict['name'] = object['name'].replace(' ', '_').replace('-', '_').replace('.', '_').replace('/', '_').lower() + '_filter'
+        macro_objects.append(new_dict)
+
+        return macro_objects
+
+    def parse_and_add_lookups(self, search_string):
+        lookups_found = re.findall('lookup (?:update=true)?(?:append=t)?\s*([^\s]*)', search_string)
+        lookup_objects = []
+        for lookup in lookups_found:
+            if lookup in self.lookups:
+                lookup_obj = self.lookups[lookup]
+                if not ('fields_list' in lookup_obj):
+                    csv_file_name = lookup_obj['filename']
+                    lookup_obj['csv_file_url'] = 'https://security-content.s3-us-west-2.amazonaws.com/lookups/' + csv_file_name
+
+                lookup_objects.append(lookup_obj)
+
+        return lookup_objects
+
+
+if __name__ == "__main__":
```

Comment:
I would include argument parsing here that:
* takes in the project repo location (you can have it default to `.`)
* takes in an output path you can have it default to`yaml_to_json`
see the following script as an example: https://github.com/splunk/security_content/blob/develop/bin/doc_gen.py#L371-L378

Reply:
> I would also refactor this code to be part of the `generate.py` https://github.com/splunk/security_content/blob/develop/bin/generate.py, alot of this logic already exists there. This can be exposed as another "product" via contentctl.py https://github.com/splunk/security_content/blob/develop/contentctl.py#L114 .. for example `--product API` and save the jsons to `dist/api` directory.

@d1vious,
As per your suggestion, this is now callable from `contentctl.py` as shown:

    `python3 contentctl.py --path . --verbose generate --product API --output dist/api`

I refactored `yaml_to_json.py` to work from within `generate.py`.  I didn't try to copy and paste the contents over into `generate.py` as there was not any good overlap of existing functionalities.  Instead, I simply modified and imported `yaml_to_json.py` as a module called from within `generate.py` and indirectly callable from `contentctl.py`.

Moreover, I also updated `yaml_to_json.py` to accept the params for repo location and output path for JSONs.  As per your suggestion, we're calling this new functionality.

For lack of imagination, the API JSONs will be uploaded to a new folder called `json` on s3/security-content, during production builds:
`aws s3 cp dist/api s3://security-content/json --recursive --exclude "*" --include "*.json"`